Epic 6798 : List Domain Designer Conversion

labkey-bpatel opened the pull request:
- Updates related to allowing binding of DomainKind specific options/properties to a bean & generifying DomainKind.

labkey-bpatel committed:
Updates related to allowing binding of DomainKind specific options/properties to a bean & generifying DomainKind.

cnathe committed:
Merge branch 'develop' into fb_list_designer_6798

labkey-jeckels reviewed (approved):
This is OK, but may not be necessary if we introduce the convenience subclass so that most other subclasses doesn't have to specify <JSONObject>

labkey-bpatel committed:
Code Review change:
Extend convenience subclass BaseAbstractDomainKind; and remove getTypeClass() from the impl classes, since its already implemented in BaseAbstractDomainKind.

cnathe committed:
Merge branch 'develop' into fb_list_designer_6798

labkey-jeckels reviewed (changes requested):
Not needed anymore, I think

labkey-bpatel commented:
Yup, you're right, no longer needed. Closing PR & will delete the fb.
